### PR TITLE
Add types for rosbag.js

### PR DESCRIPTION
**Pull request by b8s256**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/rosbag/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,508 @@
+// Type definitions for rosbag 2.6
+// Project: https://github.com/cruise-automation/rosbag.js#readme
+// Definitions by: Itthipol Lateh <https://github.com/Rovula>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.7
```
Comment (peterblazejewicz) [inline types/rosbag/index.d.ts:5]: I'd say you no longer need this on DT (this is current version enforced/tested on DT, after TS 4.4 release), unless there was a specific reason of adding this one:
```suggestion
```

**Comment by peterblazejewicz** [comment]
so this is a module with `default` export, but also with global script support:
https://github.com/cruise-automation/rosbag.js#installation
Do you want to support the syntax (no modules):
```ts
// $ExpectType Promise<Bag>
rosbag.open('file.bag');

// $ExpectType Promise<Bag>
rosbag.open(
    new File(['file'], 'file.bag', {
        type: 'text/plain',
    }),
);

// $ExpectType Promise<Bag>
rosbag.open(new Blob(['abc'], { type: 'text/plain' }));

// $ExpectType Date
rosbag.TimeUtil.toDate({ sec: 0, nsec: 0 });
```
```ts
export as namespace rosbag;
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by naeemy** [comment]
Ready to merge

**Pull request merged**